state/application.go

```diff
@@ -1337,6 +1337,46 @@ func (a *Application) GetScale() int {
 	return a.doc.DesiredScale
 }
 
+// ChangeScale alters the existing scale by the provided change amount, returning the new amount.
+// This is used on CAAS models.
+func (a *Application) ChangeScale(scaleChange int) (int, error) {
+	newScale := a.doc.DesiredScale + scaleChange
+	if newScale < 0 {
+		return a.doc.DesiredScale, errors.NotValidf("cannot remove more units than currently exist")
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			alive, err := isAlive(a.st, applicationsC, a.doc.DocID)
+			if err != nil {
+				return nil, errors.Trace(err)
+			} else if !alive {
+				return nil, applicationNotAliveErr
+			}
+			if err := a.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+			newScale = a.doc.DesiredScale + scaleChange
```

Comment:
Something to ponder - I wonder if the scale changes from underneath us, we should just error and not try again. ie if I saw an app had 5 units and I wanted to add 2 more expecting 7, and the scale changed to 4 before my txn ran, the end result would be 6 and not what i expected. Maybe it's better just to error in that case? With a suitable message. Although add-unit has the same problem so maybe not an issue to address right now